drivers/i2c/i2c_wch.c

```diff
@@ -393,14 +499,108 @@ static int i2c_wch_init(const struct device *dev)
 		return err;
 	}
 
-	config->irq_config_func(dev);
+	return 0;
+}
+
+#if defined(CONFIG_I2C_TARGET)
+
+int i2c_wch_target_register(const struct device *dev, struct i2c_target_config *config)
+{
+	const struct i2c_wch_config *cfg = dev->config;
+	struct i2c_wch_data *data = dev->data;
+	I2C_TypeDef *regs = cfg->regs;
+	int ret;
+
+	if (!config) {
+		return -EINVAL;
+	}
+
+	if (data->slave_attached) {
+		return -EBUSY;
+	}
+
+	if (data->master_active) {
+		return -EBUSY;
+	}
+
+	ret = i2c_wch_configure(dev, I2C_MODE_CONTROLLER | i2c_map_dt_bitrate(cfg->bitrate));
+	if (ret < 0) {
+		LOG_ERR("i2c: failure configure");
+		return ret;
```

Comment:
Some of this code logs on the error path and some doesn't. For consistency, do it one way or the other everywhere.

Personally I prefer not to log as the ch32v003 doesn't have much flash to spare.